library/core/src/num/nonzero.rs

```diff
@@ -287,6 +287,130 @@ nonzero_integers_div! {
     NonZeroUsize(usize);
 }
 
+macro_rules! nonzero_integers_add {
+    ( $( $Ty: ident($Int: ty); )+ ) => {
+        $(
+
+            impl $Ty {
+                /// Add a positive integer to a non-zero value.
+                /// Return None on overflow.
+                #[unstable(feature = "nonzero_add", issue = "none")]
+                #[inline]
+                pub fn checked_add(self, other: $Int) -> Option<$Ty> {
+                    self.get().checked_add(other).map(|result| {
+                        // SAFETY: $Int::checked_add returns None on overflow
+                        // so the result cannot be zero.
+                        unsafe { $Ty::new_unchecked(result) }
+                    })
+                }
+
+                /// Add a positive integer to a non-zero value.
+                /// Return maximum type value on overflow.
```

Comment:
You can use something like this instead to put a link to the max constant:

```suggestion
                /// Return
                #[doc = concat!("[`", stringify!($Int), "::MAX`]")]
                /// on overflow.
```

Reply:
Oh, so the macro variable `$Int` *can* be parsed as an element of the docstring :D Amazing, I'm adding this as well.